### PR TITLE
fix(surveys): remove a duplicate footer and fix some buggy `Cancel` button behavior

### PR DESCRIPTION
**Pull request by dmarticus**
## NOTE

I split this change out from https://github.com/PostHog/posthog/pull/22375 because the `beforeunload` stuff ended up being much more complex than I anticipated, and I still want to get the other bug fixes in before I tackle the `beforeunload` stuff.  This change just covers some small UI fixes described below.

## Problem

This PR does the following:

1. fixes the behavior for cancel on new surveys; before these changes, clicking cancel on a new survey would take you to this weird ephemeral state of this new survey. Now, clicking cancel on a new survey that hasn't been saved just takes you back to the /surveys URL. [See this message for context](https://posthog.slack.com/archives/C034XD440RK/p1716327059331599).
2. Removes the the Cancel and Save buttons from the footer. Felt unnecessary. See https://github.com/PostHog/posthog/pull/22375#discussion_r1607438057 for details.

Finally, as part of these changes, I fixed an issue where the initial survey creation process never set the isEditingSurvey state to true.

## Changes

New survey initialization experience (featuring (1) entering edit mode immediately, (2) cancel button working, (3) no footer):

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Manually tested the UI.  Performed the following tests:

- Creating a new survey from scratch immediately puts you into the editing mode
- Creating a new survey from a template immediately puts you into the editing mode
- Cancelling edits on a new survey without saving them takes you back to the surveys page (and doesn't create an unnamed survey in some vacuous state)
- Cancelling the editing process on an existing survey takes you back to that existing survey overview.

### CODE REVIEW

File: frontend/src/scenes/surveys/Survey.tsx [hunk 1]
```diff
@@ -28,7 +28,6 @@ export const scene: SceneExport = {
 
 export function SurveyComponent({ id }: { id?: string } = {}): JSX.Element {
     const { isEditingSurvey, surveyMissing } = useValues(surveyLogic)
-    const showSurveyForm = id === 'new' || isEditingSurvey
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/Survey.tsx:31]: I check this state in the `urlToAction` handler instead now, so we don't need to evaluate this logic in the component.

File: frontend/src/scenes/surveys/Survey.tsx [hunk 2]
```diff
@@ -51,6 +50,19 @@ export function SurveyForm({ id }: { id: string }): JSX.Element {
...
+            editingSurvey(false)
+            if (id === 'new') {
+                router.actions.push(urls.surveys())
+            } else {
+                loadSurvey()
+            }
+        } else {
+            router.actions.push(urls.surveys())
+        }
+    }
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/Survey.tsx:64]: This fixes the `Cancel` button issue I called out here: https://posthog.slack.com/archives/C034XD440RK/p1716327059331599

File: frontend/src/scenes/surveys/Survey.tsx [hunk 3]
```diff
@@ -88,26 +93,6 @@ export function SurveyForm({ id }: { id: string }): JSX.Element {
...
-                            router.actions.push(urls.surveys())
-                        }
-                    }}
-                >
-                    Cancel
-                </LemonButton>
-                <LemonButton type="primary" data-attr="save-survey" htmlType="submit" loading={surveyLoading}>
-                    {id === 'new' ? 'Save as draft' : 'Save'}
-                </LemonButton>
-            </div>
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/Survey.tsx:110]: Drive-by change that's not really in scope, but Boy Scout Principle and all that. I noticed that we had a footer on this page, which felt superfluous given that the page has a sticky header.  Here's what it looks liek with the footer

Screenshot 2024-05-20 at 5 07 21 PM

And what it looks like without

Screenshot 2024-05-20 at 5 06 59 PM

I prefer without.

File: frontend/src/scenes/surveys/surveyLogic.tsx [hunk 4]
```diff
@@ -756,6 +756,7 @@ export const surveyLogic = kea<surveyLogicType>([
                 urlMatchType: values.urlMatchTypeValidationError,
             }),
             submit: (surveyPayload) => {
+                actions.editingSurvey(false)
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/surveyLogic.tsx:759]: I noticed that we weren't turning off edit mode on survey submission, which I think we should since the edits are complete when the survey has been submitted.

File: frontend/src/scenes/surveys/surveyLogic.tsx [hunk 5]
```diff
@@ -770,6 +771,9 @@ export const surveyLogic = kea<surveyLogicType>([
             // This avoids resetting form fields if you click back/forward.
             if (method === 'PUSH') {
                 if (props.id) {
+                    if (props.id === 'new') {
+                        actions.editingSurvey(true)
+                    }
```
Comment (dmarticus) [inline frontend/src/scenes/surveys/surveyLogic.tsx:776]: here's the `urlToAction` change I talked about above.

File: frontend/src/scenes/surveys/Survey.tsx (hunk 2 above)
Comment (neilkakkar) [reply frontend/src/scenes/surveys/Survey.tsx:64]: small readability nit - I think this is equivalent, but let me know if I'm missing something!

```suggestion
    const handleCancelClick = (): void => {
        editingSurvey(false)
        if (id === 'new') {
            router.actions.push(urls.surveys())
        } else {
            loadSurvey()
        }
    }
```

**Review by neilkakkar — APPROVED** [review]
nice work, much better 🚀
